packages/aws-cdk-lib/aws-stepfunctions/lib/states/choice.ts

```diff
@@ -47,7 +54,7 @@ export class Choice extends State {
   public readonly endStates: INextable[] = [];
 
   constructor(scope: Construct, id: string, props: ChoiceProps = {}) {
-    super(scope, id, props);
+    super(scope, props.stateName? props.stateName: id, props);
```

Comment:
This changes the construct ID though.

So passing `stateName` now has 2 effects, affecting the construct ID as well as the state name.

It's just an override for `id` now, which is not what we want.